src/app/features/charts/components/score-barchart/score-barchart.component.ts

```diff
@@ -45,26 +45,32 @@ export class ScoreBarChartComponent implements AfterViewInit, OnChanges, OnDestr
   constructor(private helperService: HelperService) {
   }
 
-  ngOnChanges(): void {
-    if (this.initialized) {
-      // if chart has already been initialized, this means the data has changed
-      this.createChart();
+  ngOnChanges(changes: SimpleChanges): void {
+    if (
+      (changes.data && !changes.data.firstChange) ||
+      (changes.score && !changes.score.firstChange) ||
+      (changes.barColor && !changes.barColor.firstChange)
+    ) {
+      if (this.score === null) {
+        this.clearChart();
+        this.hideChart();
+      } else {
+        this.clearChart();
```

Comment:
we need to clear the chart even if the score is not null as the score may be another non-null score and hence we should redraw the chart